compiler-plugin/src/main/kotlin/arrow/meta/internal/kastree/ast/Writer.kt

```diff
@@ -80,26 +80,13 @@ open class Writer(
             children(parents, ", ")
           }
           childTypeConstraints(typeConstraints)
-          if (members.isNotEmpty()) lineEnd(" {").indented {
-            // First, do all the enum entries if there are any
-            val enumEntries = members.map { it as? Node.Decl.EnumEntry }.takeWhile { it != null }
-            enumEntries.forEachIndexed { index, enumEntry ->
-              lineBegin().also { children(enumEntry) }
-              when (index) {
-                members.size - 1 -> lineEnd()
-                enumEntries.size - 1 -> lineEnd(";").line()
-                else -> lineEnd(",")
-              }
-            }
-            // Now the rest of the members
-            childrenLines(members.drop(enumEntries.size), extraMidLines = 1)
-          }.lineBegin("}")
```

Comment:
@bloderxd I was wondering if I could get a little more context for removing this part regarding `EnumEntries`?

Reply:
ohh - I see it got moved down under ClassBody.  I'd still like to hear more about it!

Reply:
As we've created a specific `ClassBody` AST node we start to provide a separated scope to that, then what I did here was just traverse its node children instead of handle it as a `List<Decl>`, because now we have this list in a specific class body node

Reply:
@ahinchman1 As we've created a specific ClassBody AST node we start to provide a separated scope to that, then what I did here was just traverse its node children instead of handle it as a List<Decl>, because now we have this list in a specific class body node